couchbase/json-buckets.yml

```diff
@@ -0,0 +1,1071 @@
+        type: double_counter
+        label: cmd gets
+        description: Number of get commands
+
+      - name: cmd.set
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.cmd_set[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:sum()
+        type: double_counter
+        label: cmd sets
+        description: Number of set commands
+
+      - name: docs.disk.actual.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_actual_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: doc actual disk size
+        unit: bytes
+        description: Couch docs total size on disk in bytes.
+
+      - name: docs.data.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_data_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: doc data disk size
+        unit: bytes
+        description: Couch docs data size in bytes
+
+      - name: docs.disk.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: docs disk size
+        unit: bytes
+        description: Couch docs total size in bytes
+
+      - name: docs.fragmentation
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_docs_fragmentation[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
+        label: doc fragmentation
+        unit: "%"
+        description: Couch docs fragmentation
+
+      - name: data.spatial.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_spatial_data_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: gauge
+        label: spatial data size
+        unit: bytesals
+        description: Size of object data for spatial views
+
+      - name: disk.spatial.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_spatial_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: gauge
+        label: spatial disk size
+        unit: bytes
+        description: Amount of disk space occupied by spatial views
+
+      - name: ops.spatial
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_spatial_ops[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: gauge
+        label: spatial ops
+        description: Spatial operations
+
+      - name: disk.size
+        source: json:/pools/default/buckets/${bucketName}/stats?zoom=minute $.op.samples.couch_total_disk_size[0:${SPM_MONITOR_COLLECT_INTERVAL}] return:avg()
+        type: double_gauge
```

Comment:
Decimal bytes? And check the following metrics, also `double_gauge` is used. Perhaps we need avg() function that produces longs?

Reply:
Yes, let me know when we have that function.
i found few metrics, it should be long, but some time it return double such as cache hit. so i use double for safety.

Reply:
Introduced ability to cast, you should be able to use `return:long(avg())` and type `long_gauge`.

Reply:
I change to return:long(avg()) and long_gauge